students/DiannaTingg/lessons/lesson08/assignment/inventory.py

```diff
@@ -0,0 +1,63 @@
+"""
+Lesson 08 Assignment
+Functional Programming
+"""
+
+import csv
+from functools import partial
+
+# pylint: disable-msg=line-too-long
+# pylint: disable-msg=invalid-name
+
+
+def add_furniture(invoice_file="", customer_name="", item_code="", item_description="", item_monthly_price=""):
+    """
+    Updates the master invoice file which lists which furniture is rented to which customer.
+    Will create a new file if one doesn't already exist.
+    :param invoice_file: master csv file
+    :param customer_name: customer name
+    :param item_code: item code
+    :param item_description: item description
+    :param item_monthly_price: item monthly price
+    """
+
+    with open(invoice_file, "a+", newline='') as csvfile:
+        writer = csv.writer(csvfile)
+        row = customer_name, item_code, item_description, item_monthly_price
+        writer.writerow(row)
+
+
+def add_test_data():
+    """
+    Adds test data to the master invoice file.
+    """
+    add_furniture("invoice_file.csv", "Elisa Miles", "LR04", "Leather Sofa", "25.00")
+    add_furniture("invoice_file.csv", "Edward Data", "KT78", "Kitchen Table", "10.00")
+    add_furniture("invoice_file.csv", "Alex Gonzales", "BR02", "Queen Mattress", "17.00")
+
+
+def single_customer(customer_name, invoice_file):
+    """
+    Bulk processes a list of items that have been rented to a single customer.
+    :param customer_name: customer name
+    :param invoice_file: master invoice csv file
+    :return: function that iterates through rental_items and adds each item to the master invoice_file.
+    """
```

Comment:
Great doc string. Very important for potentially confusing functions. :) Probably would be helpful to be even more explicit than the instructions were, and specify that rental_items is a csv file with rows of tem_code, item_description and item_monthly_price that need to be entered into the master invoice_file.